src/Enecuum/Assets/Nodes/Routing/RoutingWorker.hs

```diff
@@ -0,0 +1,132 @@
+    let mAddress  =  findNextForHash (routingRuntime ^. myNodeAddres . A.nodeId) connects
+    whenJust mAddress $ \(_, receiverAddress) -> do
+        let privateKey  = True
+        hello <- makeRoutingHello privateKey (routingRuntime ^. myNodeAddres)
+        void $ L.notify (A.getUdpAddress receiverAddress) hello
+
+-- check if all connections are alive
+-- return list of "dead connects"
+pingConnects :: ChordRouteMap A.NodeAddress -> L.NodeL [A.NodeId]
+pingConnects nodes = do
+    deadNodes <- forM (elems nodes) $ \(nodeId, address) -> do
+        res <- L.makeRpcRequest (A.getRpcAddress address) M.Ping
+        pure $ case res of
+            Right M.Pong -> Nothing
+            Left  _      -> Just nodeId
+    pure $ catMaybes deadNodes
+
+-- sending requests to the network to clarify the map of connections
+-- successors may be better connections than those that are now.
+successorsRequest :: RoutingRuntime -> L.NodeL ()
+successorsRequest routingRuntime = do
+    -- clarify the map of connections
+    connects <- getConnects routingRuntime
+    let myNodeId = routingRuntime ^. myNodeAddres . A.nodeId
+    let loop i = do
+            let mAddress = snd <$> findInMapNByKey (\h j -> D.hashToWord64 h + 2 ^ j) i myNodeId connects
+            whenJust mAddress $ \address -> do
+                (eAddress :: Either Text A.NodeAddress) <- L.makeRpcRequest (A.getRpcAddress address) $ M.ConnectRequest myNodeId i
+                whenRight eAddress $ \(address :: A.NodeAddress) ->
+                    unless (address == routingRuntime ^. myNodeAddres) $ do
+                        L.modifyVarIO (routingRuntime ^. connectMap) (addToMap (address ^. A.nodeId) address)
+                        when (i /= 0) $ loop (i - 1)
+                whenLeft eAddress $ \_ -> when (i /= 0) $ loop (i - 1)
+    loop 63
+
+connecRequests :: Word64 -> RoutingRuntime -> L.NodeL ()
+connecRequests i routingRuntime = when (i > 0) $ do
+    let bnRpcAddress =  A.getRpcAddress (routingRuntime^.bnAddress)
+    maybeAddress <- L.makeRpcRequest bnRpcAddress $ M.ConnectRequest (routingRuntime ^. myNodeAddres . A.nodeId) i
+    case maybeAddress of
+        Right (receivedNodeId, address) | routingRuntime ^. myNodeAddres /= address -> do
+            L.modifyVarIO (routingRuntime ^. connectMap) $ addToMap receivedNodeId address
+            connecRequests (i - 1) routingRuntime
+        _ -> pure ()
+
+nextRequest :: RoutingRuntime -> L.NodeL ()
+nextRequest routingRuntime = do
+    let bnRpcAddress  =  A.getRpcAddress (routingRuntime^.bnAddress)
+    nextForMe        <- L.makeRpcRequest bnRpcAddress $ M.NextForMe (routingRuntime ^. myNodeAddres . A.nodeId)
+    case nextForMe of
+        Right (receivedNodeId, address) | routingRuntime ^. myNodeAddres /= address ->
+            L.modifyVarIO (routingRuntime ^. connectMap) (addToMap receivedNodeId address)
+        _ -> pure ()
+
+-- leave in the list of connections only those
```

Comment:
-- clear list of connects ( connect must satisfy the conditions of the chord algorithm )